tap_stripe/schemas/disputes.json

```diff
@@ -0,0 +1,205 @@
+{
+  "type": [
+    "null",
+    "object"
+  ],
+  "properties":{
+    "id": {
+      "type": [
+        "string"
+      ]
+    },
+    "object": {
+      "type": [
+        "null",
+        "string"
+      ]
+    },
+    "amount": {
+      "type": [
+        "null",
+        "integer"
+      ]
+    },
+    "balance_transaction": {
+      "type": [
+        "null",
+        "string"
+      ]
+    },
+    "balance_transactions": {
+      "type": [
+        "null",
+        "array"
+      ],
+      "items": {
```

Comment:
The elements in this array are `balance_transaction` objects. Should I list out all the keys for that, or should I use something like `"$ref" : "balance_transactions.json#/"`?

Reply:
since we have a balance transactions stream available as a top-level object, I'd say we should just keep the `id` property here so it can be joined through to the source of truth table, unless the object in this array carries any additional information

Reply:
Great, yeah that makes sense. Out of curiosity, how does an array of dicts get flattened in targets generally?

According to the docs, if there's 1 item it's "funds withdrawn" and if there's 2, the second is "funds reinstated". I wonder if it would make more sense to manually flatten this in the tap so it makes more sense for end users?

Reply:
can you link to those docs?

Reply:
https://stripe.com/docs/api/disputes/object#dispute_object-balance_transactions

Reply:
it seems like it could make sense to flatten to something like:

```
"balance_transaction_withdrawn": {
  "type": ["null", "string"]
},
"balance_transaction_reinstated": {
  "type": ["null", "string"]
}
```

Although I'm not sure if it's better to "keep true" to the source or to make the data easier to work with.

Reply:
that's always the hardest question :)

This seems like a somewhat complex transformation of the data, which introduces surface area for errors, so I'm inclined to leave it as a single `balance_transactions` array containing objects with only the `id` field of each balance transaction. The schema would be:

```
"balance_transactions": {
"type": "array",
"items": {"properties": {"id": {"type": ["null", "string"]}}}
}
```
An analyst can then join onto the `balance_transactions` data to find out what was a charge versus a credit.

Reply:
Yeah that makes sense... as I was looking at where I would add that transformation in the code it definitely seemed error prone.